### PR TITLE
[OCPCLOUD-1066] Makefile fix 

### PR DESCRIPTION
**Pull request by SamuelStuchly**
Fixed broken Makefile targets.
Added script to fix permission issues.
Updated hacking-guide.

### CODE REVIEW

File: hack/go-fmt.sh [hunk 1]
```diff
@@ -1,16 +1,7 @@
 #!/bin/sh
-if [ "$IS_CONTAINER" != "" ]; then
+
   for TARGET in "${@}"; do
```
Comment (lobziik) [inline hack/go-fmt.sh:3]: Fix indentation please.

File: Makefile [hunk 2]
```diff
@@ -120,30 +129,33 @@ unit:
...
-	hack/go-lint.sh -min_confidence 0.3 $(go list -f '{{ .ImportPath }}' ./...)
+	 $(DOCKER_CMD) hack/go-lint.sh -min_confidence 0.3 $(go list -f '{{ .ImportPath }}' ./...)
 
 .PHONY: fmt
 fmt: ## Go fmt your code
-	hack/go-fmt.sh .
+	$(DOCKER_CMD) hack/go-fmt.sh .
 
 .PHONY: goimports
 goimports: ## Go fmt your code
```
Comment (lobziik) [inline Makefile:153]: Suggest to fix this docstring.
Something like `Updates and shows diff for import lines`

File: Makefile [hunk 3]
```diff
@@ -20,13 +20,27 @@ endif
...
+endif
+
 ifeq ($(NO_DOCKER), 1)
   DOCKER_CMD =
   IMAGE_BUILD_CMD = imagebuilder
 else
-  DOCKER_CMD := docker run --env GO111MODULE=$(GO111MODULE) --env GOFLAGS=$(GOFLAGS) --rm -v "$(PWD)":/go/src/github.com/openshift/machine-api-operator:Z -w /go/src/github.com/openshift/machine-api-operator golang:1.15
-#  DOCKER_CMD := docker run --env GO111MODULE=$(GO111MODULE) --env GOFLAGS=$(GOFLAGS) --rm -v "$(PWD)":/go/src/github.com/openshift/machine-api-operator:Z -w /go/src/github.com/openshift/machine-api-operator registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.15-openshift-4.6
-  IMAGE_BUILD_CMD = docker build
+  DOCKER_CMD := $(ENGINE) run --env GO111MODULE=$(GO111MODULE) --env GOFLAGS=$(GOFLAGS) --rm -v "$(PWD)":/go/src/github.com/openshift/machine-api-operator:Z  -w /go/src/github.com/openshift/machine-api-operator openshift/origin-release:golang-1.15
```
Comment (Danil-Grigorev) [inline Makefile:41]: I would suggest leaving default `golang:1.15` image, as not everyone has access to `openshift/origin-release`

File: Makefile [hunk 4]
```diff
@@ -20,13 +20,27 @@ endif
...
+#  DOCKER_CMD := $(ENGINE) run --env GO111MODULE=$(GO111MODULE) --env GOFLAGS=$(GOFLAGS) --rm -v "$(PWD)":/go/src/github.com/openshift/machine-api-operator:Z -w /go/src/github.com/openshift/machine-api-operator registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.15-openshift-4.6
```
Comment (Danil-Grigorev) [inline Makefile:42]: We don't need it.

Comment (SamuelStuchly) [reply Makefile:42]: The commented out command ?

File: Makefile (hunk 3 above)
Comment (SamuelStuchly) [reply Makefile:41]: The reason for this change is that `golang:1.15` does not contain some packages, for example `golint`. Also providers use `openshift/origin-release:golang-1.15`. I asked @JoelSpeed  before making this change and he supported it.

File: Makefile (hunk 4 above)
Comment (SamuelStuchly) [reply Makefile:42]: Keep it as seperate commit in case we need it again ? Or can be squashed ? @Danil-Grigorev

Comment (Danil-Grigorev) [reply Makefile:42]: Just squash it. `registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.15-openshift-4.6` is an internal image-registry reference, definitely not user friendly location for pulling an image.

**Review by elmiko — APPROVED** [review]
this is all working pretty well for me with podman on fedora 33, haven't tried on docker
nicely done Sam!
/lgtm

File: Makefile (hunk 3 above)
Comment (alexander-demichev) [reply Makefile:41]: origin-release should be public

File: Makefile [hunk 5]
```diff
@@ -20,13 +20,26 @@ endif
...
+ifeq ($(USE_DOCKER), 1)
+	ENGINE=docker	
+endif
+
 ifeq ($(NO_DOCKER), 1)
   DOCKER_CMD =
   IMAGE_BUILD_CMD = imagebuilder
 else
-  DOCKER_CMD := docker run --env GO111MODULE=$(GO111MODULE) --env GOFLAGS=$(GOFLAGS) --rm -v "$(PWD)":/go/src/github.com/openshift/machine-api-operator:Z -w /go/src/github.com/openshift/machine-api-operator golang:1.15
-#  DOCKER_CMD := docker run --env GO111MODULE=$(GO111MODULE) --env GOFLAGS=$(GOFLAGS) --rm -v "$(PWD)":/go/src/github.com/openshift/machine-api-operator:Z -w /go/src/github.com/openshift/machine-api-operator registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.15-openshift-4.6
```
Comment (alexander-demichev) [inline Makefile:28]: This line should not be removed. Seems like @michaelgugino left it there on purpose https://github.com/openshift/machine-api-operator/pull/789/files#r558550502

**Comment by JoelSpeed** [comment]
Tested on mac, seems to be working for me too

File: Makefile (hunk 5 above)
Comment (michaelgugino) [reply Makefile:28]: Yes, please leave this.  We could even include another comment about why it's a comment :)

Comment (SamuelStuchly) [reply Makefile:28]: Put it back and added a comment.

**Pull request merged**